### PR TITLE
Fixed offset computation for mutant residues in protein sequence

### PR DESCRIPTION
**Pull request by iskandr**
* Added unit tests to make sure that mutant amino acids of `ProteinSequence` match our expectation

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/isovar/42)

### CODE REVIEW

File: isovar/translation.py [hunk 1]
```diff
@@ -324,13 +324,22 @@ def find_mutant_amino_acid_interval(
     # rounding down since a change in the middle of a codon should count
     # toward the variant codons
     n_coding_nucleotides_before_variant = len(cdna_coding_prefix)
-    n_complete_prefix_codons = cdna_variant_start_offset // 3
+
+    # subtract off the number of nucleotides we dropped to shift the sequence
```
Comment (ihodes) [inline isovar/translation.py]: I'm confused by the subtracting comment here; there is only integer division here.

File: isovar/translation.py [hunk 2]
```diff
@@ -324,13 +324,22 @@ def find_mutant_amino_acid_interval(
...
+    # into the correct reading frame reading frame
+    n_complete_prefix_codons = n_coding_nucleotides_before_variant // 3
 
     frame_of_variant_nucleotides = n_coding_nucleotides_before_variant % 3
     frameshift = abs(n_ref - n_alt) % 3 != 0
     indel = n_ref != n_alt
 
-    variant_aa_interval_start = n_complete_prefix_codons + 1
+    if n_ref == 0 and frame_of_variant_nucleotides == 0:
+        # if inserting at the codon boundary then add 1 to the number of
```
Comment (ihodes) [inline isovar/translation.py]: What is being inserted?

File: isovar/translation.py (hunk 1 above)
Comment (iskandr) [inline isovar/translation.py]: Old comment, thanks for catching it

File: isovar/translation.py (hunk 2 above)
Comment (iskandr) [inline isovar/translation.py]: I realized that I don't have to worry about insertions as a special case when working in base-0 offsets.

File: isovar/translation.py [hunk 3]
```diff
@@ -324,13 +324,17 @@ def find_mutant_amino_acid_interval(
...
+    # subtract off the number of nucleotides we dropped to shift the sequence
+    # into the correct reading frame reading frame
+    n_complete_prefix_codons = n_coding_nucleotides_before_variant // 3
 
     frame_of_variant_nucleotides = n_coding_nucleotides_before_variant % 3
     frameshift = abs(n_ref - n_alt) % 3 != 0
     indel = n_ref != n_alt
 
-    variant_aa_interval_start = n_complete_prefix_codons + 1
+    variant_aa_interval_start = n_complete_prefix_codons
```
Comment (ihodes) [inline isovar/translation.py]: So this was the bug?

Comment (iskandr) [inline isovar/translation.py]: And the `n_complete_prefix_codons` line above.

**Pull request merged**